File: lib/libxl-3.8.1.0/examples/fortran/vs2008/invoice/invoice.f90
program invoice

use libxl

integer i
type(BookHandle) book
type(SheetHandle) sheet, nullSheet
type(FontHandle) nullFont, boldFont, titleFont
type(FormatHandle) nullFormat, titleFormat, headerFormat, descriptionFormat, amountFormat, totalLabelFormat, totalFormat, signatureFormat

book = xlCreateXMLBook()

boldFont = xlBookAddFont(book, nullFont)
call xlFontSetBold(boldFont, 1)

titleFont = xlBookAddFont(book, nullFont)
i = xlFontSetName(titleFont, "Arial Black")
call xlFontSetSize(titleFont, 16)

titleFormat = xlBookAddFormat(book, nullFormat)
i = xlFormatSetFont(titleFormat, titleFont)

headerFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetAlignH(headerFormat, ALIGNH_CENTER)
call xlFormatSetBorder(headerFormat, BORDERSTYLE_THIN)
i = xlFormatSetFont(headerFormat, boldFont)
call xlFormatSetFillPattern(headerFormat, FILLPATTERN_SOLID)
call xlFormatSetPatternForegroundColor(headerFormat, COLOR_TAN)

descriptionFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetBorderLeft(descriptionFormat, BORDERSTYLE_THIN)

amountFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetNumFormat(amountFormat, NUMFORMAT_CURRENCY_NEGBRA)
call xlFormatSetBorderLeft(amountFormat, BORDERSTYLE_THIN)
call xlFormatSetBorderRight(amountFormat, BORDERSTYLE_THIN)

totalLabelFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetBorderTop(totalLabelFormat, BORDERSTYLE_THIN)
call xlFormatSetAlignH(totalLabelFormat, ALIGNH_RIGHT)
i = xlFormatSetFont(totalLabelFormat, boldFont)

totalFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetNumFormat(totalFormat, NUMFORMAT_CURRENCY_NEGBRA)
call xlFormatSetBorder(totalFormat, BORDERSTYLE_THIN)
i = xlFormatSetFont(totalFormat, boldFont)
call xlFormatSetFillPattern(totalFormat, FILLPATTERN_SOLID)
call xlFormatSetPatternForegroundColor(totalFormat, COLOR_YELLOW)

signatureFormat = xlBookAddFormat(book, nullFormat)
call xlFormatSetAlignH(signatureFormat, ALIGNH_CENTER)
call xlFormatSetBorderTop(signatureFormat, BORDERSTYLE_THIN)

sheet = xlBookAddSheet(book, "Invoice", nullSheet)

i = xlSheetWriteStr(sheet, 2, 1, "Invoice No. 3568", titleFormat)
i = xlSheetWriteStr(sheet, 4, 1, "Name: John Smith", nullFormat)
i = xlSheetWriteStr(sheet, 5, 1, "Address: San Ramon, CA 94583 USA", nullFormat)
i = xlSheetWriteStr(sheet, 7, 1, "Description", headerFormat)
i = xlSheetWriteStr(sheet, 7, 2, "Amount", headerFormat)
i = xlSheetWriteStr(sheet, 8, 1, "Ball-Point Pens", descriptionFormat)
i = xlSheetWriteNum(sheet, 8, 2, 85, amountFormat)
i = xlSheetWriteStr(sheet, 9, 1, "T-Shirts", descriptionFormat)
i = xlSheetWriteNum(sheet, 9, 2, 150, amountFormat)
i = xlSheetWriteStr(sheet, 10, 1, "Tea cups", descriptionFormat)
i = xlSheetWriteNum(sheet, 10, 2, 45, amountFormat)
i = xlSheetWriteStr(sheet, 11, 1, "Total:", totalLabelFormat)
i = xlSheetWriteFormula(sheet, 11, 2, "=SUM(C9:C11)", totalFormat)
i = xlSheetWriteStr(sheet, 14, 2, "Signature", signatureFormat)

i = xlSheetSetCol(sheet, 1, 1, 40, nullFormat, 0);
i = xlSheetSetCol(sheet, 2, 2, 15, nullFormat, 0);

i = xlBookSave(book, "invoice.xlsx")

call xlBookRelease(book)

end program invoice

File: lib/libxl-3.8.1.0/examples/c++/vs2012/invoice/invoice.cpp
#include <iostream>
#include <windows.h>
#include "libxl.h"

using namespace libxl;

int main()
{
    Book* book = xlCreateXMLBook();
    if(book) 
    {   
        Font* boldFont = book->addFont();
        boldFont->setBold();

        Font* titleFont = book->addFont();
        titleFont->setName(L"Arial Black");
        titleFont->setSize(16);

        Format* titleFormat = book->addFormat();
        titleFormat->setFont(titleFont);

        Format* headerFormat = book->addFormat();
        headerFormat->setAlignH(ALIGNH_CENTER);
        headerFormat->setBorder(BORDERSTYLE_THIN);
        headerFormat->setFont(boldFont);        
        headerFormat->setFillPattern(FILLPATTERN_SOLID);
        headerFormat->setPatternForegroundColor(COLOR_TAN);

        Format* descriptionFormat = book->addFormat();
        descriptionFormat->setBorderLeft(BORDERSTYLE_THIN);

        Format* amountFormat = book->addFormat();
        amountFormat->setNumFormat(NUMFORMAT_CURRENCY_NEGBRA);
        amountFormat->setBorderLeft(BORDERSTYLE_THIN);
        amountFormat->setBorderRight(BORDERSTYLE_THIN);
                
        Format* totalLabelFormat = book->addFormat();
        totalLabelFormat->setBorderTop(BORDERSTYLE_THIN);
        totalLabelFormat->setAlignH(ALIGNH_RIGHT);
        totalLabelFormat->setFont(boldFont);

        Format* totalFormat = book->addFormat();
        totalFormat->setNumFormat(NUMFORMAT_CURRENCY_NEGBRA);
        totalFormat->setBorder(BORDERSTYLE_THIN);
        totalFormat->setFont(boldFont);
        totalFormat->setFillPattern(FILLPATTERN_SOLID);
        totalFormat->setPatternForegroundColor(COLOR_YELLOW);

        Format* signatureFormat = book->addFormat();
        signatureFormat->setAlignH(ALIGNH_CENTER);
        signatureFormat->setBorderTop(BORDERSTYLE_THIN);
             
        Sheet* sheet = book->addSheet(L"Invoice");
        if(sheet)
        {
            sheet->writeStr(2, 1, L"Invoice No. 3568", titleFormat);

            sheet->writeStr(4, 1, L"Name: John Smith");
            sheet->writeStr(5, 1, L"Address: San Ramon, CA 94583 USA");

            sheet->writeStr(7, 1, L"Description", headerFormat);
            sheet->writeStr(7, 2, L"Amount", headerFormat);

            sheet->writeStr(8, 1, L"Ball-Point Pens", descriptionFormat);
            sheet->writeNum(8, 2, 85, amountFormat);
            sheet->writeStr(9, 1, L"T-Shirts", descriptionFormat);
            sheet->writeNum(9, 2, 150, amountFormat);
            sheet->writeStr(10, 1, L"Tea cups", descriptionFormat);
            sheet->writeNum(10, 2, 45, amountFormat);

            sheet->writeStr(11, 1, L"Total:", totalLabelFormat);
            sheet->writeFormula(11, 2, L"=SUM(C9:C11)", totalFormat);

            sheet->writeStr(14, 2, L"Signature", signatureFormat);

            sheet->setCol(1, 1, 40);
            sheet->setCol(2, 2, 15);
          }

          if(book->save(L"invoice.xlsx")) 
          {
            ::ShellExecute(NULL, L"open", L"invoice.xlsx", NULL, NULL, SW_SHOW);
          }

          book->release();   
    }

    return 0;
}
